docs/src/api.md

```diff
@@ -13,8 +14,9 @@ NoInverse
 InverseFunctions.test_inverse
 ```
 
-## Additional functions
+## Additional functionality
 
 ```@docs
 InverseFunctions.square
+InverseFunctions.FunctionWithInverse
```

Comment:
Since it's not exported and not intended to be constructed directly, I wonder if it should be omitted from the docs?

Reply:
I thought people might want to have the option to dispatch on it, in low-level uses cases and so on, that's why I made it party of the official API.

Reply:
I assumed that this was the motivation. I wonder, however, how often one would actually want to dispatch on `FunctionWithInverse`. I assume for many cases it would be simpler to just use `unwrap_f` (or something equivalent of the same name) and `inverse` - similar to the `setinverse` case in this PR.

Reply:
We may actually have occasion to do that in the pushforward measure in MeasureBase.

Reply:
>  since PushforwardMeasure stores both forward and inverse function for performance reasons

To obtain these you don't have to specialize on `FunctionWithInverse` though. You could just store `_unwrap_f(f)` and `inverse(f)` if an arbitrary `f` is provided.